File: codility/lesson1-iterations/BinaryGap.py
'''
Lesson1 - BinaryGap
https://app.codility.com/programmers/lessons/1-iterations/binary_gap/
'''
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(N):
    # write your code in Python 3.6
    binary = bin(N)[2:]
    cnt = binary.count('0')
    if cnt == 0:
        return 0
    while cnt > 0:
        if '1' + '0' * cnt + '1' in binary:
            return cnt
        else:
            cnt -= 1
    if cnt == 0:
        return 0

File: codility/lesson1-iterations/BinaryGap.cpp
'''
Lesson1 - BinaryGap
https://app.codility.com/programmers/lessons/1-iterations/binary_gap/
'''
	
int solution(int N) {
    // write your code in C++14 (g++ 6.2.0)
    int MaxGap = 0;
    int GapCount = 0;
    int flag = 0;
    while (N > 0) {
        if ((N & 1) == 1 and flag == 1) {
            MaxGap = std::max(MaxGap, GapCount);
            GapCount = 0;
        }
        else if ((N & 1) == 1 and flag == 0) {
            flag = 1;
        }
        else if ((N & 1) != 1 and flag == 1) {
            GapCount += 1;
        }
        N >>= 1;
    }
    return MaxGap;
}
